client/scenes/stream_tracking.cpp

```diff
@@ -138,6 +142,25 @@ void scenes::stream::tracking()
 #ifdef __ANDROID__
 	// Runtime may use JNI and needs the thread to be attached
 	application::instance().setup_jni();
+	jni::object<""> act(application::native_app()->activity->clazz);
+	auto app = act.call<jni::object<"android/app/Application">>("getApplication");
+	auto ctx = app.call<jni::object<"android/content/Context">>("getApplicationContext");
+
+	jni::string filter_jstr("android.intent.action.BATTERY_CHANGED");
+	jni::string level_jstr("level");
+	jni::string scale_jstr("scale");
+	jni::string plugged_jstr("plugged");
+	jni::Int default_jint(-1);
+
+	auto receiver_jobj = jni::object<"android/content/BroadcastReceiver">(NULL);
+	auto filter_jobj = jni::new_object<"android/content/IntentFilter">(filter_jstr);
+
+	auto register_receiver = jni::klass("android/content/Context")
+	                                 .method<jni::object<"android/content/Intent">>("registerReceiver", receiver_jobj, filter_jobj);
+	auto get_int_extra = jni::klass("android/content/Intent").method<jni::Int>("getIntExtra", level_jstr, default_jint);
```

Comment:
while this works, i think it would be more proper to pass the actual typenames, except i'm not sure how to do that.